Uniswap/interface
# fix: handle invalid chainId on position page

Makes sure that being on an unsupported chain does not crash the /pools/{positionId} pages.

The error is caused here: https://github.com/Uniswap/interface/blob/main/src/constants/tokens.ts#L492

I don't think we should change the behavior there to not throw an error, because there are other users of that method. We would need to test all of them. Similarly, returning undefined or mainnet ether is not reasonable behavior in my opinion. It should be a precondition that we are verified to be on a supported chain before getting to that call.

I need to separate the Invalid state into a separate component so the wrap hook isn't called before we check if it's an unsupported chain. This is a rule of hooks that we can't conditionally call them.

## Description
_Slack thread:_ https://uniswapteam.slack.com/archives/C047U65H422/p1681232054878529

## Screen capture

after:
invalid position on supported chain
<img width="1395" alt="Screen Shot 2023-04-11 at 12 04 51 PM" src="https://user-images.githubusercontent.com/4899429/231249610-df62edba-c6b5-4bce-8387-e1caba99c80a.png">

invalid position on unsupported chain
<img width="1196" alt="Screen Shot 2023-04-11 at 12 52 06 PM" src="https://user-images.githubusercontent.com/4899429/231249619-67f1cc79-4bbd-440f-8e53-1d80322ed744.png">

valid position on supported chain
<img width="1197" alt="Screen Shot 2023-04-11 at 12 52 13 PM" src="https://user-images.githubusercontent.com/4899429/231249627-2d9c4b6e-bcf7-457a-960d-033f52f72f85.png">

before:
<img width="1199" alt="Screen Shot 2023-04-11 at 2 02 46 PM" src="https://user-images.githubusercontent.com/4899429/231250328-f73dff6a-5122-4d9f-8c41-c10219790d08.png">


## Test plan
### QA (ie manual testing)
- Go to /pools/12401824 and then switch to Sepolia in MetaMask
- Should see "go back to pools", not a crash

#### Devices
I tested on desktop Chrome.

### Automated testing
- [x] Unit test (n/a)
- [x] Integration/E2E test (n/a -- this is not doable without work to integrate Hardhat with Cypress, as we need to switch to an unsupported network to test this)
